api/desecapi/models.py

```diff
@@ -169,18 +178,26 @@ class Token(rest_framework.authtoken.models.Token):
     name = models.CharField("Name", max_length=64, default="")
     user_specific_id = models.BigIntegerField("User-Specific ID")
 
+    class Meta:
+        abstract = False
+        unique_together = (('user', 'user_specific_id'),)
+
+    def generate_key(self):
+        plain = b64encode(urandom(21)).decode('utf-8').replace('/', '-').replace('=', '_').replace('+', '.')
+        return TokenHashWithPlaintext(plain)
+
+    @property
+    def plain(self):
+        try:
+            return self.key.plain
```

Comment:
seems unnecessarily complicated: just `return self.plain`?

Reply:
will go away if the `TokenHashWithPlaintext` discussion is settled

Reply:
gone